pkg/xray/resource_xray_workers_count.go

```diff
@@ -0,0 +1,280 @@
+		"analysis": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: "The number of workers involved in scanning analysis.",
+			Elem: &schema.Resource{
+				Schema: newExistingContentSchema,
+			},
+		},
+		"impact_analysis": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: "The number of workers involved in Impact Analysis to determine how a component with a reported issue impacts others in the system.",
+			Elem: &schema.Resource{
+				Schema: newContentSchema,
+			},
+		},
+		"notification": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			MaxItems:    1,
+			Description: "The number of workers managing notifications.",
+			Elem: &schema.Resource{
+				Schema: newContentSchema,
+			},
+		},
+	}
+
+	type NewContent struct {
+		New int `json:"new_content"`
+	}
+
+	type NewExistingContent struct {
+		NewContent
+		Existing int `json:"existing_content"`
+	}
+
+	/*
+		API returns the follow JSON structure:
+		{
+		    "index": {
+		        "new_content": 4,
+		        "existing_content": 2
+		    },
+		    "persist": {
+		        "new_content": 4,
+		        "existing_content": 2
+		    },
+		    "analysis": {
+		        "new_content": 4,
+		        "existing_content": 2
+		    },
+		    "alert": {
+		        "new_content": 4,
+		        "existing_content": 2
+		    },
+		    "impact_analysis": {
+		        "new_content": 2
+		    },
+		    "notification": {
+		        "new_content": 2
+		    }
+		}
+	*/
+	type WorkersCount struct {
+		Index          NewExistingContent `json:"index"`
+		Persist        NewExistingContent `json:"persist"`
+		Analysis       NewExistingContent `json:"analysis"`
+		Alert          NewExistingContent `json:"alert"`
+		ImpactAnalysis NewContent         `json:"impact_analysis"`
+		Notification   NewContent         `json:"notification"`
+	}
+
+	var newHclContentConstructor = func(src interface{}) map[string]interface{} {
+		return map[string]interface{}{
+			"new_content": src.(NewContent).New,
+		}
+	}
+
+	var newExistingHclContentConstructor = func(src interface{}) map[string]interface{} {
+		return map[string]interface{}{
+			"new_content":      src.(NewExistingContent).New,
+			"existing_content": src.(NewExistingContent).Existing,
+		}
+	}
+
+	var packContent = func(d *schema.ResourceData, attr string, src interface{}, hclContentConstructor func(src interface{}) map[string]interface{}) []error {
+		setValue := util.MkLens(d)
+
+		resource := workersCountSchema[attr].Elem.(*schema.Resource)
+		content := hclContentConstructor(src)
+		return setValue(attr, schema.NewSet(schema.HashResource(resource), []interface{}{content}))
+	}
+
+	var packWorkersCount = func(d *schema.ResourceData, workersCount WorkersCount) diag.Diagnostics {
+		var errors []error
+
+		errors = append(errors, packContent(d, "index", workersCount.Index, newExistingHclContentConstructor)...)
```

Comment:
You don't need this append - `packContent` already returns `setValue` 

just do
```go
packContent(...)
errors := packContent(...) // last call
```

Reply:
```go
// setValue is now a closure that is maintaining the current running list of errors and applying it to 'd'
setValue = util.MkLens(d) 
errors := setValue("hello", "world")
errors = setValue("foo", baz)
```
the last assignment and invocation always contains the most recent set of errors. The closure appends for you. So, when you do as you're doing, you're duplicating function. It works just like it would in javascript

```go
func MkLens(d *schema.ResourceData) Lens {
	var errors []error
	return func(key string, value interface{}) []error {
		if err := d.Set(key, value); err != nil {
			errors = append(errors, err)
		}
		return errors
	}
}
```